modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -383,9 +383,14 @@ public static String getSimpleRef(String ref) {
             //throw new RuntimeException("Failed to get the schema: null");
             return null;
         } else if (ref.startsWith("#/components/")) {
-            ref = ref.substring(ref.lastIndexOf("/") + 1);
+
+            ref = ref.substring("#/components/".length()) ;
+            // allow deep-linking, only remove first ref in components (schemas, pathitems, ...)
+            if (ref.contains("/")) {
+                ref = ref.substring(ref.indexOf('/') + 1 );
+            }
         } else if (ref.startsWith("#/definitions/")) {
-            ref = ref.substring(ref.lastIndexOf("/") + 1);
+            ref = ref.substring("#/definitions/".length());
```

Comment:
Openapi 3.1.0 spec processing is not yet supported. Definitions is a 3.1.0 feature

Reply:
So should i remove this part? definitions where part of the util before ... i just aligned the logic

Reply:
Yes you should, its inclusion is a bug